pallets/staking-rewards/src/lib.rs

```diff
@@ -380,6 +397,53 @@ pub mod pallet {
 
             InflationInfo { inflation, roi }
         }
+
+        /// The account ID of the rent rewards pool.
+        pub fn rent_pool_account_id() -> T::AccountId {
+            use sp_runtime::traits::TrailingZeroInput;
+
+            let entropy =
+                (T::PalletId::get(), b"gear rent pool").using_encoded(sp_io::hashing::blake2_256);
+            let actor_id = Decode::decode(&mut TrailingZeroInput::new(entropy.as_ref()))
+                .expect("infinite length input; no invalid inputs for type; qed");
+
+            actor_id
+        }
+
+        /// Return the amount in the rent pool.
+        // The existential deposit is not a part of the pool so the account never gets deleted.
+        pub fn rent_pool_balance() -> BalanceOf<T> {
+            T::Currency::free_balance(&Self::rent_pool_account_id())
+                // Must never be less than 0 but better be safe.
+                .saturating_sub(T::Currency::minimum_balance())
+        }
+    }
+}
+
+// TODO: consider to optimize the process #3729
+fn pay_rent_rewards_out<T: Config>(maybe_active_era_info: Option<ActiveEraInfo>) {
+    let Some(active_era_info) = maybe_active_era_info else {
+        return;
+    };
+
+    let reward_points = pallet_staking::Pallet::<T>::eras_reward_points(active_era_info.index);
+    let total = u128::from(reward_points.total);
+    if total == 0 {
+        return;
+    }
+
+    let funds: u128 = pallet::Pallet::<T>::rent_pool_balance().unique_saturated_into();
+    for (account_id, points) in reward_points.individual {
+        let payout = funds.saturating_mul(u128::from(points)) / total;
```

Comment:
I have a gut feeling that integer saturating math is not the best way of finding a fraction of the total (prone to rounding error). Especially having multiplication ahead of division. Fixed-point arithmetic seems to be a better candidate, does it not? Or why not use some `PerThing` trait types (like `Perbill` or even `Perquintill`) like we do for inflation numbers calculation (or in staking pallet itself)?

Reply:
I had the same on my mind while reviewing but it works well, any excess tokens gonna be redistributed in the next era so not looks like an issue

Reply:
Using `PerThing` types is seamless: `Perquintill::from_rational(points, total) * funds`, but it'll guarantee better accuracy. Anyway, since it's not that a big deal I'm not insisting.